ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/resources/impl/ResourceManager.java

```diff
@@ -433,7 +434,7 @@ public File apply(Resource[] resources) throws FunctionException {
                     return update(path, createRequest);
                 }
 
-                return ps.createProject(dto).thenPromise(new Function<ProjectConfigDto, Promise<Project>>() {
+                return ps.createProject(dto, ((MutableProjectConfig)createRequest.getBody()).getOptions()).thenPromise(new Function<ProjectConfigDto, Promise<Project>>() {
```

Comment:
Reeeeally long and hard to understand line of code.
It would be nice to move `((MutableProjectConfig)createRequest.getBody()).getOptions()` to the local variable.
Agree?